### ISSUE TITLE
add support for Hugo front-matter

### ISSUE THREAD

**Issue by M-M-M-M** [MacDownApp/macdown#935]
Add support for Hugo front-matter in TOML.

Identified by opening and closing +++.

so a document with some content between 
```
+++

my meta data from front-matter here

+++

# Here is the real markdown to be displayed

and only this ;-)

```

**Comment by uranusjr** [comment]
So… it’s like Jekyll front matter, but instead of `---` I use `+++`, and instead of YAML the content is parsed as TOML. Am I correct?

**Comment by M-M-M-M** [comment]
you are correct.

**Comment by uranusjr** [comment]
Cool.

**Comment by franklinyu** [comment]
So we need a TOML parser for this? Like [libtoml](https://github.com/ajwans/libtoml)?

**Comment by uranusjr** [comment]
Yeah. There are a few choices (JavaScript would work, but C is probably better). `toml-rs` is likely the best if we are willing to add Rust to the build (probably not) because it’s very heavily used in Cargo.

For reference: https://github.com/toml-lang/toml/wiki#implementations

**Comment by drakeguan** [comment]
Quite happy to see this proposal, and I'd like to support it too.
